src/main/java/com/github/takayoshi24/cinema/movie/MovieController.java

```diff
@@ -1,40 +1,52 @@
 package com.github.takayoshi24.cinema.movie;
 
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.UUID;
+
+@RequiredArgsConstructor
 @RestController
+@RequestMapping("/movies")
 public class MovieController {
 
     private final MovieService movieService;
 
-    @Autowired
-    public MovieController(MovieService movieService) {
-        this.movieService = movieService;
+
+    @PostMapping
+    public MovieDTO registerNewMovie(@Valid @RequestBody MovieCreateDTO movieData){
+        Movie movie = movieService.addNewMovie(movieData);
+        return new MovieDTO(movie);
+    }
+
+    @GetMapping
+    public Page<MovieDTO> getAllMovies(Pageable pageable){
+        Page<Movie> movies = movieService.getAllMovies(pageable);
+        return movies.map(MovieDTO::new);
     }
 
+
     @GetMapping(params = {"genre"})
     public Page<Movie> getMovieByGenre(@RequestParam String  genre, Pageable pageable){
         return movieService.findAllByGenre(genre,pageable);
     }
 
-    @PostMapping(path="/movie")
-    public void registerNewMovie(@RequestBody Movie movie){
-        movieService.addNewMovie(movie);
-    }
 
-    @DeleteMapping(path = "{movieId}")
-    public void deleteMovie(@PathVariable("movieId") Long movieId){
-        movieService.deleteMovie(movieId);
+    @PutMapping(path = "/{movieId}")
+    public MovieDTO updateMovie(
+            @PathVariable("movieId") UUID movieId,
+            @RequestBody MovieCreateDTO movieData){
+        Movie movie = movieService.updateMovie(movieId, movieData.genre(), movieData.title());
```

Comment:
Everytime MovieCreateDTO changes, you'd have to change updateMovie() method too.
To avoid it, change the signature to `Movie updateMovie(UUID movieId, MovieCreateDTO dto);`.